### PR TITLE
Allow rewrite status to be 2xx or 4xx.

### PR DESCRIPTION
**Pull request by ax-nathan**
Matt, first thanks for the Caddy project...a true showcase of Golang's strengths!

I have a case where it would be useful to do the following:

```
# Headfake the editor
rewrite /editor/spec {
  if {method} is PUT
  status 200
}
```

However, the current implementation prevents this use case, which I think is safe and reasonable. This PR allow status to be 2xx or 4xx. _Tests are included_. I hope you agree that this simple change is OK. Thanks again!

### CODE REVIEW

**Comment by mholt** [comment]
Thanks for the tests.

What's your use case? We specifically limited this to 4xx to prevent misuse of this directive.

**Comment by ax-nathan** [comment]
I suppose you could consider this a "misuse"...but useful! 

I'm proxying the Swagger Editor as part of a development sandbox effort. The browser JS wants to PUT the content of the editor for safekeeping on the backend, but we are using the editor in readonly mode, loading static content, so we want to prevent the PUT. Unfortunately, a failed PUT breaks the JS on the client. So, in summary, this is a case of using a proxy to adapt off-the-shelf components to an unanticipated use case. Make sense?

**Comment by ax-nathan** [comment]
P.S. I tried to sign the CLA, but it seems to be confused by the difference between my two GH accounts...one that did the commit, and on that did the PR. :-(

**Comment by mholt** [comment]
What if you just put an empty file in that path instead?

**Comment by ax-nathan** [comment]
It's a PUT...which returns a 405.

File: caddy/setup/rewrite_test.go [hunk 1]
```diff
@@ -154,6 +159,22 @@ func TestRewriteParse(t *testing.T) {
...
+			status 200
+		 }`, false, []rewrite.Rule{
+			&rewrite.ComplexRule{Base: "/", Regexp: regexp.MustCompile(".*"), Status: 200},
+		}},
+		{`rewrite {
+			to /to
+			status 200
+		 }`, false, []rewrite.Rule{
+			&rewrite.ComplexRule{Base: "/", To: "/to", Regexp: regexp.MustCompile(".*"), Status: 200},
+		}},
```
Comment (mholt) [inline caddy/setup/rewrite_test.go]: Call me pedantic, but I think these tests don't need the regexp, right? If not, maybe we could remove it to keep the case more obvious.

**Comment by mholt** [comment]
Okay, I'm convinced, this is definitely an edge case, but a harmless enough change as far as I can tell.

Note my inline comment and then sign the CLA with the matching email address and we're good to go I think.

**Comment by ngConsulti** [comment]
CLA signed.

**Comment by mholt** [comment]
Thank you!
